packages/expo-screen-orientation/ios/EXScreenOrientation/EXScreenOrientationRegistry.m

```diff
@@ -0,0 +1,38 @@
+// Copyright 2019-present 650 Industries. All rights reserved.
+
+
+#import "EXScreenOrientationRegistry.h"
+#import <UMCore/UMDefines.h>
+
+@implementation EXScreenOrientationRegistry
+
+
+UM_REGISTER_SINGLETON_MODULE(EXScreenOrientationRegistry)
+
+- (void)setOrientationMask:(UIInterfaceOrientationMask)orientationMask
+          forAppId:(NSString *)appId
+{
+  NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
+  [userDefaults setInteger:orientationMask forKey:appId];
+  [userDefaults synchronize];
+}
+
+- (UIInterfaceOrientationMask)getOrientationMaskForAppId:(NSString *)appId
+{
+  NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
+  if (@([userDefaults integerForKey:appId]) == nil) {
+    return UIInterfaceOrientationMaskAllButUpsideDown;
+  }
+    return [userDefaults integerForKey:appId];
+}
+
+- (BOOL)doesKeyExistForAppId:(NSString *)appId
+{
+  NSUserDefaults *userDefaults = [NSUserDefaults standardUserDefaults];
+  if (@([userDefaults integerForKey:appId]) == nil) {
```

Comment:
You're getting an integer, casting it to the object to check whether it's nil - this looks weird.
```suggestion
  if ([userDefaults objectForKey:appId]) {
```